File: myapplication_ui/src/services/DriverAssignmentService.js
import http from "../http-common"; 

class DriverAssignmentService {
  getAllDriverAssignments(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/driverAssignment/driverAssignments`, searchDTO);
  }

  get(driverAssignmentId) {
    return this.getRequest(`/driverAssignment/${driverAssignmentId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/driverAssignment?field=${matchData}`, null);
  }

  addDriverAssignment(data) {
    return http.post("/driverAssignment/addDriverAssignment", data);
  }

  update(data) {
  	return http.post("/driverAssignment/updateDriverAssignment", data);
  }
  
  uploadImage(data,driverAssignmentId) {
  	return http.postForm("/driverAssignment/uploadImage/"+driverAssignmentId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new DriverAssignmentService();

File: src/main/java/com/myapplication/service/DriverAssignmentService.java
package com.myapplication.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.myapplication.domain.DriverAssignment;
import com.myapplication.dto.DriverAssignmentDTO;
import com.myapplication.dto.DriverAssignmentSearchDTO;
import com.myapplication.dto.DriverAssignmentPageDTO;
import com.myapplication.dto.DriverAssignmentConvertCriteriaDTO;
import com.myapplication.service.GenericService;
import com.myapplication.dto.common.RequestDTO;
import com.myapplication.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface DriverAssignmentService extends GenericService<DriverAssignment, Integer> {

	List<DriverAssignment> findAll();

	ResultDTO addDriverAssignment(DriverAssignmentDTO driverAssignmentDTO, RequestDTO requestDTO);

	ResultDTO updateDriverAssignment(DriverAssignmentDTO driverAssignmentDTO, RequestDTO requestDTO);

    Page<DriverAssignment> getAllDriverAssignments(Pageable pageable);

    Page<DriverAssignment> getAllDriverAssignments(Specification<DriverAssignment> spec, Pageable pageable);

	ResponseEntity<DriverAssignmentPageDTO> getDriverAssignments(DriverAssignmentSearchDTO driverAssignmentSearchDTO);
	
	List<DriverAssignmentDTO> convertDriverAssignmentsToDriverAssignmentDTOs(List<DriverAssignment> driverAssignments, DriverAssignmentConvertCriteriaDTO convertCriteria);

	DriverAssignmentDTO getDriverAssignmentDTOById(Integer driverAssignmentId);







}





